web/static/js/components/folders/FolderShow.jsx

```diff
@@ -138,13 +180,25 @@ class FolderShow extends Component<any, any> {
 }
 
 const mapStateToProps = (state, ownProps) => {
-  const id = parseInt(ownProps.params.folderId)
-
+  const { params, t } = ownProps
+  const { projectId } = params
+
+  const folderId = params.folderId && parseInt(params.folderId)
+  const panelSurveyId = params.panelSurveyId && parseInt(params.panelSurveyId)
+  if (!folderId && !panelSurveyId) throw new Error(t('Missing param: folderId or panelSurveyId'))
+  let panelSurvey = null
+  if (state.panelSurvey && panelSurveyId == state.panelSurvey.id) {
+    panelSurvey = state.panelSurvey
+  }
   // Right now we show all surveys: they are not paginated nor sorted
   let surveys = state.surveys.items
 
   if (surveys) {
-    surveys = values(surveys).filter(s => s.folderId == id)
+    if (panelSurvey) {
+      surveys = values(surveys).filter(s => panelSurvey && s.panelSurveyOf == panelSurvey.id)
+    } else {
+      surveys = values(surveys).filter(s => s.folderId == folderId)
+    }
```

Comment:
a little comment here wouldn't hurt 🙃  (and extra points for extracting this along with the others assignments to `surveys` that are bellow)

If you like it, here are some functions (may have errors since I haven't been writing something in js for a long time)

```javascript
const filterSurveys = (allSurveys, panelSurvey, folder) => {
  const values = allSurveys ? values(allSurveys) : []
  return if (panelSurvey) {
    // If a panelSurvey is present, then we should filter those survey which have the same parent
    values.filter(s => panelSurvey && s.panelSurveyOf == panelSurvey.id)
  } else {
    // Is a common folder, so we should filter those surveys that are inside of it
    values.filter(s => s.folderId == folder.id)
  }
}

const sortAndSliceSurveys = (surveys, pageIndex, pageSize) => {
  // Sort by updated at, descending
  const sorted = (surveys || []).sort((x, y) => y.updatedAt.localeCompare(x.updatedAt))
  // Show only the current page
  return values(sorted).slice(pageIndex, pageIndex + pageSize)
}

```